pandas/tests/indexes/test_range.py

```diff
@@ -1037,3 +1037,27 @@ def test_engineless_lookup(self):
             assert idx.get_loc("a") == -1
 
         assert "_engine" in idx._cache
+
+    def test_reindex_limit(self):
+        # GH 28631
+        data = [["A", "A", "A"], ["B", "B", "B"], ["C", "C", "C"], ["D", "D", "D"]]
+        exp_data = [
+            ["A", "A", "A"],
+            ["B", "B", "B"],
+            ["C", "C", "C"],
+            ["D", "D", "D"],
+            ["D", "D", "D"],
+            [np.nan, np.nan, np.nan],
+        ]
+        df = pd.DataFrame(data)
+        result = df.reindex([0, 1, 2, 3, 4, 5], method="ffill", limit=1)
+        expected = pd.DataFrame(exp_data)
+        tm.assert_frame_equal(result, expected)
+
+    def test_get_indexer_limit(self):
+        # GH 28631
+        idx = Index(range(4))
+        target = Index([0, 1, 2, 3, 4, 5])
```

Comment:
should we explicitly create a `RangeIndex` here and above?

Reply:
yeah, actually this file is to test `RangeIndex`